flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -390,51 +395,101 @@ private void callSelect(SqlCommandCall cmdCall) {
 			view.open();
 
 			// view left
-			terminal.writer().println(CliStrings.messageInfo(CliStrings.MESSAGE_RESULT_QUIT).toAnsi());
-			terminal.flush();
+			printInfo(CliStrings.MESSAGE_RESULT_QUIT);
 		} catch (SqlExecutionException e) {
-			printException(e);
+			printExecutionException(e);
 		}
 	}
 
 	private boolean callInsertInto(SqlCommandCall cmdCall) {
-		terminal.writer().println(CliStrings.messageInfo(CliStrings.MESSAGE_SUBMITTING_STATEMENT).toAnsi());
-		terminal.flush();
+		printInfo(CliStrings.MESSAGE_SUBMITTING_STATEMENT);
 
 		try {
 			final ProgramTargetDescriptor programTarget = executor.executeUpdate(context, cmdCall.operands[0]);
 			terminal.writer().println(CliStrings.messageInfo(CliStrings.MESSAGE_STATEMENT_SUBMITTED).toAnsi());
 			terminal.writer().println(programTarget.toString());
 			terminal.flush();
 		} catch (SqlExecutionException e) {
-			printException(e);
+			printExecutionException(e);
 			return false;
 		}
 		return true;
 	}
 
-	private void callSource(SqlCommandCall cmdCall) {
-		final String pathString = cmdCall.operands[0];
+	private void callCreateView(SqlCommandCall cmdCall) {
+		final String name = cmdCall.operands[0];
+		final String query = cmdCall.operands[1];
+
+		try {
+			// validate with a copy
+			final SessionContext contextCopy = context.copy();
```

Comment:
This check fails if the VIEW query contains UDFs. 
It seems that these are not registered yet.